cpp/src/slice2cs/CsUtil.cpp

```diff
@@ -85,52 +52,18 @@ Slice::CsGenerator::getNamespacePrefix(const ContainedPtr& cont)
 string
 Slice::CsGenerator::getNamespace(const ContainedPtr& cont)
 {
-    string scope = fixId(cont->scope());
-    if (scope.rfind('.') == scope.size() - 1)
-    {
-        scope = scope.substr(0, scope.size() - 1);
-    }
-    string prefix = getNamespacePrefix(cont);
-    if (!prefix.empty())
-    {
-        if (!scope.empty())
-        {
-            return prefix + "." + scope;
-        }
-        else
-        {
-            return prefix;
-        }
-    }
-
-    return scope;
-}
+    assert(!dynamic_pointer_cast<Module>(cont));
 
-string
-Slice::CsGenerator::getUnqualified(const string& type, const string& scope, bool builtin)
-{
-    if (type.find('.') != string::npos && type.find(scope) == 0 && type.find('.', scope.size() + 1) == string::npos)
-    {
-        return type.substr(scope.size() + 1);
-    }
-    else if (builtin)
-    {
-        return type.find('.') == string::npos ? type : "global::" + type;
-    }
-    else
-    {
-        return "global::" + type;
-    }
+    string scope = cont->mappedScope(".").substr(1);
```

Comment:
So mapped scope returns `".Foo.Bar."`, and that is why we need `substr(1)` here?

I wonder if we can have a scope prefix, and C# can pass either "" or "global::" and use it for the initial occurrence.

Reply:
Yes, that's correct, and also not a bad idea.

But I want to check how often we actually use this. Because the other option is just to remove this global scope piece all together and let callers add it by hand. Because I'm pretty sure we almost always just remove it except for niche things like type-ids.

Reply:
It would make the intent clear, otherwise each time you see this code have to wonder why is `substr(1)` necessary.

Reply:
I think we should either:
 - don't start with the scope in the string returned by mappedScope (and update mappedScope doc-comment)
 or
 - add a prefix parameter

Reply:
I fully agree with @bernardnormier's comment.

All I'm saying is that I lean towards not starting with the scope separator, but I want to wait and see how we're actually using this. I suspect the leading separator is more of a nuisance than a help, but I won't know until I look.